modules/optimizing-by-encapsulation.adoc

```diff
@@ -0,0 +1,35 @@
+// Module included in the following assemblies:
+//
+// * scalability_and_performance/optimizing-networking.adoc
+
+[id="optimizing-cpu-usage_{context}"]
+= Encapsulating mount namespaces
+
+Mount namespaces are used to isolate mount points so that processes in different namespaces cannot view each others' files. Encapsulation is the process of moving mount namespaces to an alternative location, known to the owner but hidden from intrusive visitors.
+
+The host operating system uses `systemd` to constantly scan all mount namespaces: both the standard Linux mounts and the numerous mounts that Kubernetes uses to operate. The current implementation of Kubelet and CRI-O both use the top-level namespace for all container and Kubelet mountpoints. However, encapsulating these container-specific mountpoints in a private namespace reduces `systemd` overhead with no difference in functionality. Using a separate mount namespace for both CRI-O and Kubelet can encapsulate container-specific mounts from any `systemd` or other host OS interaction.
+
+This ability to potentially achieve major CPU optimization is now available to all {product-title} administrators. Encapsulation can also improve security, by storing Kubernetes-specific mount points in a location safe from inspection by unprivileged users.
+
+== How encapsulation works
+The following diagrams illustrate a Kubernetes installation before and after encapsulation. Both scenarios show example containers which have mount propagation settings of bidirectional, host-to-container, and none.
```

Comment:
Would be useful to expand on the concept of mount propagation
Which containers have which type etc

Reply:
Maybe just a link to the kubernetes documentation would be better than rehashing the whole thing here? https://kubernetes.io/docs/concepts/storage/volumes/#mount-propagation

Reply:
Will add a reference to the topic of mount propagation.

Reply:
I feel there still is a disconnect between what is written in this section and the diagrams
I.e. I  can not easily follow how the text translates into the diagrams in the preview

Reply:
I think the diagram updates will make it clearer; they were just finished yesterday and haven't been updated yet.

Reply:
With Jim's help, there's been a rewrite of the narrative. When the diagram is complete, we can do another pass.

Reply:
The finished diagrams are in place now.